src/libs/actions/Session/index.js

```diff
@@ -565,6 +565,11 @@ function authenticatePusher(socketID, channelName, callback) {
     });
 }
 
+function updateSessionLoginAndResendValidationLink(login) {
+    Onyx.merge(ONYXKEYS.CREDENTIALS, {login});
```

Comment:
We should move this to the optimistData in resendValidationLink - https://github.com/Expensify/App/blob/44a6876023facd8c0620424c9538aef676e6ee05/src/libs/actions/Session/index.js#L91-L99